src/hooks/useIdOS.ts

```diff
@@ -0,0 +1,30 @@
+import { idOS } from '@idos-network/idos-sdk';
+import { useCallback, useEffect } from 'react';
+
+import { useIdosStore } from '@/stores/idosStore';
+
+export function useIdOS() {
+  const setIdosStore = useIdosStore((state) => state.set);
+
+  const init = useCallback(async () => {
+    await new Promise<void>(async (resolve, reject) => {
```

Comment:
I don't think we need this wrapping promise: `await new Promise<void>(async (resolve, reject) => {...})`. The wrapping `useCallback(async () => {...})` already turns this function body into a promise.

Reply:
After chatting with @shelegdmitriy, we agreed we can refactor this at a later time.